examples/platform/esp32/time/TimeSync.h

```diff
@@ -0,0 +1,46 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+#include <esp_sntp.h>
+#include <lib/core/CHIPError.h>
+#include <lib/support/Span.h>
+
+namespace chip {
+class Esp32TimeSync
```

Comment:
please add class documentation in comments

Reply:
Could you add more? A class name `ESP32TimeSync` clearly adds time synchronization. The only new bit is using NTP.

When looking at this header I wonder:
  - must this be a singleton or not? If a singleton who creates it?
  - is it valid to have multiple Init calls?
  - Init seems to never be able to fail ... how? What happens if I provide bad arguments (e.g. 0 days for sync, bad SNTP server names) for example?
  - what is a `sntp server name` ? DNS name? example would be great.

Reply:
And specifically: this is a class with no members ... so why is it a class at all? It seems it should be just a namespace.

Do you expect to pass pointers around for test injection or something else?

Reply:
Removed class and used namespace.